src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -1345,4 +1345,10 @@ static constexpr Property<Affinity> affinity{"AFFINITY"};
  */
 static constexpr Property<std::vector<std::string>, PropertyMutability::RO> execution_devices{"EXECUTION_DEVICES"};
 
+/**
+ * @brief Path to the file with model's weights.
+ *
+ * @note This property is used for weightless caching. Only used when ov::CacheMode Property is set to "OPTIMIZE_SIZE".
+ */
+static constexpr Property<std::string, PropertyMutability::RW> weights_path{"WEIGHTS_PATH"};
```

Comment:
why is it public property? Is external user supposed to set it?

Can it be moved to https://github.com/openvinotoolkit/openvino/blob/master/src/inference/dev_api/openvino/runtime/internal_properties.hpp ?

Reply:
Yes, it's not saved in the cache, so the external user has to provide it like here: https://github.com/openvinotoolkit/openvino/pull/25731/files#diff-305779c23a13e2f28656783813ddbdbd8b962d20e3ec05683e52ebe89d0df259R63.

Reply:
As @vladimir-paramuzov and @slyalin wrote, we need to patch frontends to add this attribute to `ov::Model`, so users don't need to deal with this attribute.

Reply:
Yes, I'm working on this right now. This will eliminate the need to provide it to compile_model() by user. However, the user will still have to provide it to import_model() once they intend to load the model from cache.